File: backend_old/app/runtime/launchers/opengrep_launcher.py
#!/usr/bin/env python3
from __future__ import annotations

import os
import sys


for key in ("HTTP_PROXY", "HTTPS_PROXY", "ALL_PROXY", "http_proxy", "https_proxy", "all_proxy"):
    os.environ.pop(key, None)
os.environ["NO_PROXY"] = "*"
os.environ["no_proxy"] = "*"

os.execv("/usr/local/bin/opengrep.real", ["opengrep.real", *sys.argv[1:]])

File: backend/src/bin/opengrep_launcher.rs
use std::{env, os::unix::process::CommandExt, process::Command};

const PROXY_KEYS: &[&str] = &[
    "HTTP_PROXY",
    "HTTPS_PROXY",
    "ALL_PROXY",
    "http_proxy",
    "https_proxy",
    "all_proxy",
];

fn main() {
    for key in PROXY_KEYS {
        env::remove_var(key);
    }

    env::set_var("NO_PROXY", "*");
    env::set_var("no_proxy", "*");

    let err = Command::new("/usr/local/bin/opengrep.real")
        .args(env::args_os().skip(1))
        .exec();
    eprintln!("failed to exec /usr/local/bin/opengrep.real: {err}");
    std::process::exit(127);
}
